pylint/checkers/variables.py

```diff
@@ -370,6 +402,30 @@ def _ignored_modules(self):
         return get_global_option(self, 'ignored-modules', default=[])
 
 
+    @check_messages('reused-outer-loop-variable')
+    def visit_for(self, node):
+        assigned_to = _for_loop_assign_names(node.target)
+
+        # Only check variables that are used
+        dummy_rgx = self.config.dummy_variables_rgx
+        assigned_to = [var for var in assigned_to if not dummy_rgx.match(var)]
+
+        for variable in assigned_to:
+            for outer_for, outer_variables in self._loop_variables:
+                if (variable in outer_variables
+                        and not in_for_else_branch(outer_for, node)):
+                    self.add_message(
+                        'reused-outer-loop-variable',
+                        args=variable,
+                        node=node
+                    )
+                    break
+
+        self._loop_variables.append((node, assigned_to))
+
+    def leave_for(self, _):
```

Comment:
I think if reused-outer-loop-variable is disabled, then this will crash, since visit_for will never be called. Can you check this please?